### PR TITLE
Updated Cake, added bootstrappers, fixed docs and tests

### PR DESCRIPTION
**Pull request by devlead**
- Updated Cake to 0.6.4
- Fixed tests to compile with 0.6.4
- Added missing nuget/reference to Common in tests
- Added bootstrappers for easier builds for new "cloners"
- Fixed missing docs
- Adjusted build.cake to work with boostrapper defaults

### CODE REVIEW

**Comment by Redth** [comment]
Can you fix up the conflicts and I'll merge?

Thanks

**Comment by devlead** [comment]
@Redth missed a few GH notifications, this still relevant? If so I'll revisit to update PR.

**Comment by Redth** [comment]
you tell me :)

File: Cake.FileHelpers.Tests/Cake.FileHelpers.Tests.csproj [hunk 1]
```diff
@@ -28,16 +28,22 @@
...
+    <Reference Include="Cake.Common, Version=0.17.0.0, Culture=neutral, processorArchitecture=MSIL">
+      <HintPath>..\packages\Cake.Common.0.17.0\lib\net45\Cake.Common.dll</HintPath>
+      <Private>True</Private>
+    </Reference>
+    <Reference Include="Cake.Core, Version=0.17.0.0, Culture=neutral, processorArchitecture=MSIL">
+      <HintPath>..\packages\Cake.Core.0.17.0\lib\net45\Cake.Core.dll</HintPath>
+      <Private>True</Private>
+    </Reference>
+    <Reference Include="Cake.Testing, Version=0.15.2.0, Culture=neutral, processorArchitecture=MSIL">
+      <HintPath>..\packages\Cake.Testing.0.15.2\lib\net45\Cake.Testing.dll</HintPath>
```
Comment (devlead) [inline Cake.FileHelpers.Tests/Cake.FileHelpers.Tests.csproj]: Cake 0.17.0 has a dependency issue, so need to use older version until v 0.18.0

**Comment by devlead** [comment]
@Redth rebased, conflicts resolved.

**Comment by devlead** [comment]
@Redth hold this one I'll update to Cake.Testing 17.1 shortly

**Comment by devlead** [comment]
@Redth So done, ready for review.

**Pull request merged**